js/data/line_bucket.js

```diff
@@ -175,8 +181,8 @@ LineBucket.prototype.addLine = function(vertices, join, cap, miterLimit, roundLi
             var prevSegmentLength = currentVertex.dist(prevVertex);
             if (prevSegmentLength > 2 * sharpCornerOffset) {
                 var newPrevVertex = currentVertex.sub(currentVertex.sub(prevVertex)._mult(sharpCornerOffset / prevSegmentLength)._round());
-                distance += newPrevVertex.dist(prevVertex);
-                this.addCurrentVertex(newPrevVertex, flip, distance, prevNormal.mult(1), 0, 0, false);
+                this.distance += newPrevVertex.dist(prevVertex) * LINE_DISTANCE_SCALE;
```

Comment:
Do we have to scale the distance here? Or can that be done [here](https://github.com/mapbox/mapbox-gl-js/pull/2114/files#diff-7ffec697472833439424ba5814cb0c00R71)? My intent was for all buffer packing craziness to go there.